lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -526,56 +526,83 @@ static bool sinkCode(Function *F) {
       node->getNthResult(ArithmeticNode::ResultIdx).replaceAllUsesOfWith(newTR);
     }
 
+    // Sink Transpose Below RescaleQuantized.
+    // Potentially exposes opportunity to be combined up with Convolution.
+    // If it doesn't work out it will be re-sinked later.
+    if (auto *RQ = dyn_cast<RescaleQuantizedNode>(node)) {
+      auto *TR = dyn_cast<TransposeNode>(RQ->getInput());
+      if (!TR) {
+        continue;
+      }
+
+      auto newRQType = F->getParent()->uniqueTypeWithNewShape(
+          RQ->getResult().getType(), TR->getInput().getType()->dims());
+      auto *newRQ =
+          F->createRescaleQuantized(RQ->getName(), TR->getInput(), newRQType);
+      auto *newTR = F->createTranspose(TR->getName(), newRQ, TR->getShuffle());
+      RQ->getResult().replaceAllUsesOfWith(newTR);
+      changed = true;
+    }
+
     // Sink RELU below batch concat nodes.
     if (auto *CN = dyn_cast<ConcatNode>(node)) {
-      if (CN->getInputs().size() != 2) {
-        continue;
+      llvm::SmallVector<NodeValue, 6> CNInputs;
+      bool isAllRelu = true;
```

Comment:
nit: we can remove all those flags and just check CNInputs.size() against CN->getInputs().
Same for other checks.

Reply:
done.